From 37744a7b6c1fb47213bd9db9a0eb442b6ab77b05 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Thu, 1 Oct 2020 13:12:49 +0200
Subject: [PATCH] phosh: Drop rotate-display

The rotate_display served us well but it's too simplistic (no events,
degrees instead of transforms, ...). Since phosh now uses
wlr-output-management for bid to use this part of the private protocol.
---
 src/phosh.c | 37 ++-----------------------------------
 1 file changed, 2 insertions(+), 35 deletions(-)

diff --git a/src/phosh.c b/src/phosh.c
index 635b337..d9f1517 100644
--- a/src/phosh.c
+++ b/src/phosh.c
@@ -43,41 +43,8 @@ phosh_rotate_display (struct wl_client   *client,
                       struct wl_resource *surface_resource,
                       uint32_t            degrees)
 {
-  struct phosh_private *phosh = wl_resource_get_user_data (resource);
-  enum wl_output_transform transform = WL_OUTPUT_TRANSFORM_NORMAL;
-
-  g_debug ("rotation: %d", degrees);
-  degrees %= 360;
-
-  switch (degrees) {
-  case 0:
-    transform =  WL_OUTPUT_TRANSFORM_NORMAL;
-    break;
-  case 90:
-    transform = WL_OUTPUT_TRANSFORM_90;
-    break;
-  case 180:
-    transform = WL_OUTPUT_TRANSFORM_180;
-    break;
-  case 270:
-    transform = WL_OUTPUT_TRANSFORM_270;
-    break;
-  default:
-    wl_resource_post_error (resource,
-                            PHOSH_PRIVATE_ERROR_INVALID_ARGUMENT,
-                            "Can only rotate in 90 degree steps");
-  }
-
-  /* Make sure we rotate clockwise */
-  phoc_utils_fix_transform(&transform);
-
-  if (!phosh->panel) {
-    g_warning ("Tried to rotate inexistent panel");
-    return;
-  }
-
-  wlr_output_set_transform (phosh->panel->output, transform);
-  output_damage_whole(phosh->panel->output->data);
+  wl_resource_post_error (resource, WL_DISPLAY_ERROR_INVALID_OBJECT,
+                          "Use wlr-output-management protocol instead");
 }
 
 
-- 
2.26.2

